pkg/controller/account/iam.go

```diff
@@ -0,0 +1,185 @@
+package account
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/go-logr/logr"
+	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
+	"github.com/openshift/aws-account-operator/pkg/awsclient"
+	"github.com/openshift/aws-account-operator/pkg/credentialwatcher"
+)
+
+// Type for JSON response from Federation end point
+type awsSigninTokenRepsonse struct {
+	SigninToken string
+}
+
+// RequestSigninToken makes a HTTP request to retrieve a Signin Token from the federation end point
+func RequestSigninToken(reqLogger logr.Logger, awsclient awsclient.Client, DurationSeconds *int64, FederatedUserName *string, PolicyArns []*sts.PolicyDescriptorType, STSCredentials *sts.AssumeRoleOutput) (string, error) {
+
+	// // URLs for building Federated Signin queries
+	federationEndPointURL := "https://signin.aws.amazon.com/federation"
```

Comment:
Should these three be constants ?

Reply:
Possibly, I was specifically trying to keep these within the function but no reason they couldn't be moved out in the future.